### DISCUSSION TITLE
ODM support

### DISCUSSION THREAD

**Discussion by davidcorreia** [phalcon/cphalcon#15865] [General]
will phalcon 4 or 5 have ODM support any time soon (exemple for mongodb)?

it's heart breaking to keep projects with phalcon 3 :'( since the php versions are getting way to old

thank you,

**Comment by niden** [comment]
The problem we have with Mongo is the interfaces. In order to use the latest functionality of Mongo, we need to include in our code one of the interfaces provided from the Mongo extension. If we do that, it will make Phalcon **require** the mongodb extension to be present in the system and loaded before Phalcon (the same way PSR does now) whether you want to use it or not.

The code is ready to be quite honest but we have not found a good way to incorporate the two, where users that do not want to use mongo do not have to use it. 

There are plans for Mongo for our v6 version for sure. IF we do manage to find a good solution, even as a separate extension, we will also add it for v5 but that will take some time after v5 is released.

We definitely have not forgotten about this.

There is also this repository that might help for now: https://github.com/phalcon/incubator-mongodb

**Reply by davidcorreia** [reply]
thank you for the reply.

i have seen that repo, but when i try to add it via composer it says it requires the "normal" incubator 

    composer require phalcon/incubator-mongodb
    
      Problem 1
        - phalcon/incubator v3.4.6 requires ext-phalcon ^3.3 -> the requested PHP extension phalcon has the wrong version (4.1.2) installed.
        - phalcon/incubator v3.4.6 requires ext-phalcon ^3.3 -> the requested PHP extension phalcon has the wrong version (4.1.2) installed.
        - Installation request for phalcon/incubator (installed at v3.4.6, required as ^3.4) -> satisfiable by phalcon/incubator[v3.4.6].

and the normal incubator requires phalcon 3.4 :/ (or am i doing something wrong?)

**Reply by Jeckerson** [reply]
@davidcorreia It seems that you have `phalcon/incubator` inside your `composer.json` file.

**Reply by davidcorreia** [reply]
@Jeckerson you were rigth...
i just needed to remove the vendor folder as well. i will test and give feedback later
thank you :)

**Reply by davidcorreia** [reply]
@Jeckerson sorry for the trouble but could you give me some lights on how to setup the service for multiple connections?

       public function initialize()
       {
            $this->setConnectionService('ConService');
            $this->setSource('CollectionName');
       }

had to install mongodb/mongodb with composer to be able to use MongoDB/Client but still not working

     $client = new MongoDB\Client('mongodb://localhost:27017');
     return $client;

 PHP Fatal error:  Uncaught ArgumentCountError: Too few arguments to function MongoDB\\Client::selectCollection(), 1 passed in /full/path/vendor/phalcon/incubator-mongodb/src/Mvc/Collection.php on line 1390 and at least 2 expected in /full/path/vendor/mongodb/mongodb/src/Client.php:323

also tried with the driver manager but it returns "Call to undefined method MongoDB\\Driver\\Manager::selectCollection()"
